scipy/stats/tests/test_mstats_basic.py

```diff
@@ -201,6 +201,14 @@ def test_spearmanr(self):
               0.0, 0.6, 6.7, 3.8, 1.0, 1.2, 1.4, np.nan]
         (x, y) = (ma.fix_invalid(x), ma.fix_invalid(y))
         assert_almost_equal(mstats.spearmanr(x,y)[0], 0.6887299)
+        # Next test is to make sure calculation uses sufficient precision.
+        # The denominator's value is ~n^3 and used to be represented as an
+        # int. 2000**3 > 2**32 so these arrays would cause overflow on
+        # some machines.
+        x = np.arange(2000, dtype=np.float)
+        y = np.arange(2000, dtype=np.float)
+        y = np.concatenate((y[1000:], y[:1000]))
+        assert_almost_equal(mstats.spearmanr(x,y)[0], -0.50000037)
```

Comment:
We can calculate this number "by hand" by noting that the way x and y are constructed, the difference in ranks for all 2000 elements will be 1000. In other words, using the notation from wikipedia (https://en.wikipedia.org/wiki/Spearman's_rank_correlation_coefficient), the `d_i` are all 1000.

Thus, six times the sum of each `d_i` squared is 6 * 2000 * (1000^2) = 12000000000. The denominator is 2000 * (2000^1 - 1) = 7999998000.

Treating as a mixed fraction and pulling out a couple powers of 10 and a 3, we get 1 + (1333334 / 2666666). We can go further and pull out an extra 1/2 from this, giving us 1 + 1/2 + 1/2666666.

To calculate Spearman's rho we do one minus that value, giving us -1/2 - 1/2666666 as our final answer. This -1/2666666 is approximately 3.75e-7:

```python
>>> 1 / 2666666.
3.7500009375002346e-07
```

Now, that's all well and good but all I really want/need to check here is that the returned value is in the `[-1,1]` region (it returns ~21.341 before this fix on my machine). Checking for the approximately correct answer is more robust against different kinds of precision problems, since it's possible the numbers could overflow in such a way that the returned value is back inside the region `[-1,1]`, but that seems extremely unlikely.

What seems _more_ likely is that someone in the future could add checks to this function to make sure the output is within the bounds `[-1,1]` as is currently done for the Pearson calculation:

```python
    # Presumably, if r > 1, then it is only some small artifact of floating
    # point arithmetic.
    r = min(r, 1.0)
    r = max(r, -1.0)
```

If someone did the same thing for the Spearman calculation then suddenly checking the bounds `[-1, 1]` would always succeed and this would be a non-functional regression test.